code/modules/research/mechanic/mechanic_designs.dm

```diff
@@ -23,13 +24,27 @@
 		if(M.component_parts && M.component_parts.len)
 			category = "Machines"
 			for(var/obj/item/I in M.component_parts) //fetching the circuit by looking in the parts
-				if(istype(I, /obj/item/weapon/circuitboard))
-					var/obj/item/weapon/circuitboard/CB = I
-					req_tech = ConvertReqString2List(CB.origin_tech) //our tech is the circuit's requirement
+				if(I.mech_flags != MECH_SCAN_FAIL)
```

Comment:
You need to check if it has this flag, not just if it is exactly that flag; for example, imagine a scan has both MECH_SCAN_FAIL and MECH_SCAN_ILLEGAL, this wot NOT exactly match MECH_SCAN_FAIL.

I also suggest you reverse this check to avoid rightward creep (that is, if unsatisfactory, continue). For example...

````
if(I.mech_flags & MECH_SCAN_FAIL)
   continue
```